Flagsmith.Client.Test/FlagsmithClientTest.cs

```diff
@@ -0,0 +1,84 @@
+﻿using FlagsmithEngine.Environment.Models;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Moq;
+using System.Net.Http;
+using System.Threading;
+using Newtonsoft.Json;
+using System.Linq;
+
+namespace Flagsmith.FlagsmithClientTest
+{
+    internal class FlagsmithClientTest : FlagsmithClient
```

Comment:
As above, do we _need_ this class?

Reply:
we can not test private methods directly in dotnet, that is why we keep private method as protected and calling the same function from their derived class.

Reply:
I'm not sure why we need to test any private methods though? I don't think the python client tests any private methods does it?